src/MeshKernel/CurvilinearGridOrthogonalization.cpp

```diff
@@ -67,19 +65,97 @@ void meshkernel::CurvilinearGridOrthogonalization::Compute()
     }
 
     // Compute orthogonalization bounding box
-    m_minM = std::min(mFirstNode, mSecondNode);
-    m_minN = std::min(nFirstNode, nSecondNode);
-    m_maxM = std::max(mFirstNode, mSecondNode);
-    m_maxN = std::max(nFirstNode, nSecondNode);
+    const auto [minM, minN, maxM, maxN] = OrderCoordinates(mFirstNode, nFirstNode, mSecondNode, nSecondNode);
+    m_minM = minM;
+    m_minN = minN;
+    m_maxM = maxM;
+    m_maxN = maxN;
+}
+
+std::tuple<size_t, size_t, size_t, size_t> meshkernel::CurvilinearGridOrthogonalization::OrderCoordinates(size_t firstPointM, size_t firstPointN, size_t secondPointM, size_t secondPointN) const
+{
+    return {std::min(firstPointM, secondPointM), std::min(firstPointN, secondPointN), std::max(firstPointM, secondPointM), std::max(firstPointN, secondPointN)};
+}
+
+void meshkernel::CurvilinearGridOrthogonalization::SetFrozenLine(Point const& firstPoint, Point const& secondPoint)
+{
+    // Get the m and n indices from the point coordinates
+    auto [mFirstNode, nFirstNode] = m_grid->GetNodeIndices(firstPoint);
+    auto [mSecondNode, nSecondNode] = m_grid->GetNodeIndices(secondPoint);
+
+    // Coinciding nodes, no valid line, nothing to do
+    if (mFirstNode == mSecondNode && nFirstNode == nSecondNode)
+    {
+        return;
+    }
+
+    // The selected nodes must be on the vertical or horizontal line
+    const auto [minM, minN, maxM, maxN] = OrderCoordinates(mFirstNode, nFirstNode, mSecondNode, nSecondNode);
+    auto const deltaMNewLine = maxM - minM;
+    if (auto const deltaNNewLine = maxN - minN; deltaMNewLine != 0 && deltaNNewLine != 0)
+    {
+        throw std::invalid_argument("CurvilinearGridOrthogonalization::SetFrozenLine the points of the line to freeze must lie on the same grid line");
+    }
+
+    // The start-end coordinates of the new line, along m or n direction
+    auto const startNewLine = deltaMNewLine == 0 ? minN : minM;
+    auto const endNewLine = deltaMNewLine == 0 ? maxN : maxM;
+    auto const constantCoordinateLine = deltaMNewLine == 0 ? minM : minN;
+
+    // Frozen lines cannot cross existing frozen lines
+    for (auto const& frozenLine : m_frozenLines)
```

Comment:
I'm not sure, but it looks like this for-loop is for the purpose of checking consistency, and not for changing the state of anything. If that is the case you could consider making a separate function for it to reduce the function complexity a bit